react/src/components/organisms/Header/index.js

```diff
@@ -1,35 +1,119 @@
+          bodyClass = bodyClass.split(' ').filter((val) => val !== 'show-menu').join(' ');
+        } else {
+          bodyClass += ' show-menu';
+        }
+        body.setAttribute('class', bodyClass);
+      } else {
+        bodyClass = 'show-menu';
+        body.setAttribute('class', 'show-menu');
+      }
+    } else {
+      bodyClass = 'show-menu';
+      body.setAttribute('class', 'show-menu');
+    }
+    const newState = (bodyClass) ? { utilNavOpen: false } : { utilNavOpen: true };
+    if (afterButtonSearch) {
+      this.setState(newState, this.afterButtonSearch);
     } else {
-      this.setState({ utilNavOpen: true });
+      this.setState(newState);
+    }
+  };
+  // Puts focus on the mobile search input after the menu is opened via the search button.
+  afterButtonSearch = () => {
+    document.getElementById('nav-search').focus();
+    if (is.fn(this.props.headerSearch.buttonSearch.onClick)) {
+      this.props.headerSearch.buttonSearch.onClick();
+    }
+  };
+  // On click action used for both top and bottom buttons.
+  defaultButtonSearchOnClick = (e) => {
+    if (e && e.currentTarget === this.buttonRefTop.current) {
+      if (this.state.shouldNavigateTop) {
+        const query = this.searchInputTop.current.value;
+        if (query.length > 0) {
+          window.location.href = `https://search.mass.gov/?q=${query}`;
+        }
+      } else {
+        // Only toggle the menu for the top input if it's hidden (on mobile).
+        const computedStyle = getComputedStyle(this.searchInputTop.current, null);
+        if (computedStyle.display === 'none') {
+          this.menuButtonClicked(true);
+        }
+      }
     }
+    if (e && e.currentTarget === this.buttonRefBottom.current) {
+      if (this.state.shouldNavigateBottom) {
+        const query = this.searchInputBottom.current.value;
+        if (query.length > 0) {
+          // @todo - this does not work in iFrame tags. Change this to something that does.
+          window.location.href = `https://search.mass.gov/?q=${query}`;
+        }
+      } else {
+        this.menuButtonClicked(false);
+      }
+    }
+  }
+  handleChangeSearchTop = () => {
+    const shouldNavigateTop = (this.searchInputTop.current.value.length > 0);
+    this.setState({ shouldNavigateTop });
   }
+  handleChangeSearchBottom = () => {
+    const shouldNavigateBottom = (this.searchInputBottom.current.value.length > 0);
+    this.setState({ shouldNavigateBottom });
+  }
+  topHeaderSearch = () => {
+    const headerSearchProps = clone(this.props.headerSearch);
+    headerSearchProps.buttonSearch.setButtonRef = this.buttonRefTop;
+    headerSearchProps.buttonSearch.onClick = this.defaultButtonSearchOnClick;
+    headerSearchProps.inputRef = this.searchInputTop;
+    headerSearchProps.onChange = this.handleChangeSearchTop;
+    return(<HeaderSearch {...headerSearchProps} />);
+  };
+  bottomHeaderSearch = () => {
+    const headerSearchProps = clone(this.props.headerSearch);
```

Comment:
i think you can simplify lines 106-111 like so:
```
    const headerSearchProps = {
      ...this.props.headerSearch,
      buttonSearch: {
        setButtonRef: this.buttonRefBottom,
        onClick: this.defaultButtonSearchOnClick
      },
      inputRef: this.searchInputBottom,
      onChange: this.handleChangeSearchBottom,
      id: 'nav-search'
    };
```

Reply:
That will mean onSubmit would still be pointing to the original this.props.headerSearch.onSubmit, and it won't handle all potential props in this.props.orgDropdown or this.props.postInputFilter.